src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -79,17 +79,26 @@ public class PhysicalFilesWatcher : IDisposable
             ExclusionFilters filters)
         {
             _root = root;
-            _fileWatcher = fileSystemWatcher;
-            _fileWatcher.IncludeSubdirectories = true;
-            _fileWatcher.Created += OnChanged;
-            _fileWatcher.Changed += OnChanged;
-            _fileWatcher.Renamed += OnRenamed;
-            _fileWatcher.Deleted += OnChanged;
-            _fileWatcher.Error += OnError;
+
+            if (fileSystemWatcher != null)
+            {
+                _fileWatcher = fileSystemWatcher;
+                _fileWatcher.IncludeSubdirectories = true;
+                _fileWatcher.Created += OnChanged;
+                _fileWatcher.Changed += OnChanged;
+                _fileWatcher.Renamed += OnRenamed;
+                _fileWatcher.Deleted += OnChanged;
+                _fileWatcher.Error += OnError;
+            }
 
             PollForChanges = pollForChanges;
             _filters = filters;
 
+            if (fileSystemWatcher == null && !pollForChanges)
```

Comment:
(nit) why do this arg checking in the middle of the ctor? Why not do it at the top of the method?

Reply:
I guess I felt it was more of an incompatible check rather than a guard.  I suppose if this type did any argument checking I would have put it with that, so I can move.  What's your thought on the use of `ArgumentNullException` for this case?  Previously it was a NullRef.

Reply:
I think the `ArgumentNullException` makes sense. If this `bool` is `false`, then this arg can't be `null`...